snakemake/subworkflow.py

```diff
@@ -0,0 +1,66 @@
+import os
+
+from snakemake.logging import logger
+from snakemake.exceptions import WorkflowError
+
+
+def subworkflow(
+    path,
+    basedir,
+    input,
+    output,
+    params,
+    wildcards,
+    threads,
+    resources,
+    log,
+    config,
+    rulename,
+    conda_env,
+    container_img,
+    singularity_args,
+    env_modules,
+    bench_record,
+    jobid,
+    bench_iteration,
+    cleanup_scripts,
+    shadow_dir,
+    workflow,
+):
+    # extract (and normalize) parameters
+    basedir = os.path.normpath(basedir)
+    sub_workdir = os.path.normpath(params.get("workdir", f"subworkflow_{rulename}"))
+    sub_configfile = params.get("configfile", "")
+
+    # the sub-workflow needs to run in a different directory than the
+    # main-workflow. This is necessary due to Snakemake's locking mechanism
+    if basedir == sub_workdir:
+        raise WorkflowError(
+            "Sub-workflow workdir must be different from main-workflow workdir"
```

Comment:
consider including the two directories in the message

Reply:
Also, you control the `sub_workdir` so should this ever happen (isn't it a bug if it does?)

Reply:
The user can specify the workdir by setting `params.workdir` in the respective rule.
So checking this still seems sensible (at least in the current approach).